Sources/SwiftProtobuf/Message.swift

```diff
@@ -67,6 +81,12 @@ public protocol Message: CustomDebugStringConvertible {
   /// semantics for messages with heap storage; it should only be called on
   /// newly-created messages or messages where the storage has been ensured
   /// unique.
+  ///
+  /// - Parameters:
+  ///   - decoder: a `Decoder`; the `Message` will call the method
+  ///     corresponding to the type of this field.
+  ///   - protoFieldNumber: the number of the field to decode.
+  /// - Throws: an error on failure or type mismatch.
   mutating func decodeField<D: Decoder>(decoder: inout D, fieldNumber: Int) throws
```

Comment:
I believe this is meant to be internal to the library, and I thought @allevato was hoping to maybe even remove it.